docs/mgiration.md

```diff
@@ -0,0 +1,152 @@
+# Migration from RKE1 to RKE2
+
+In order to migrate from RKE to RKE 2, you need basically two things:
+
+- ETCD data directory
+- Cluster CA certificates
+
+Both can be found when you take a RKE1 snapshot. When taking a snapshot in RKE1 an archive file will be created that contains two things, the etcd snapshot and `.rkestate` of your cluster.
+
+## Introducing Migration Agent
+
+migration-agent is a tool that lays the groundwork for RKE1 nodes to move to RKE2 nodes. It accomplishes this in two main steps:
+
+- Restore the etcd snapshot on etcd nodes to the RKE2 etcd db data directory.
+- Copy the CA certs and service account token key from `.rkestate` file to RKE2 data directory.
+
+This tool runs on RKE1 nodes before running RKE2.
+
+### Usage
+
+1- To run the migration you need to take a snapshot first on rke1 nodes:
+
+```
+rke etcd snapshot-save --s3 --name rke1snapshot --access-key <access-key> --secret-key <secret-key> --region <region> --folder <folder> --bucket-name <bucket name>
+```
+
+For more information, please refer to the RKE1 [official documentation](https://rancher.com/docs/rke/latest/en/etcd-snapshots/one-time-snapshots/)
+
+2- Now you can either run mgiration agent directly on the node, or use the following [mainfest](https://github.com/rancher/migration-agent/blob/master/deploy/daemonset.yaml) to deploy migration-agent as a daemonset on the RKE1 cluster. Before you run the manifest file you need to edit the file to include the information about the s3 snapshot of RKE1:
+```
+command:
+          - "sh"
+          - "-c"
+          - "migration-agent --s3-region <region> --s3-bucket hgalal-s3 --s3-folder <s3 folder> --s3-access-key <access-key> --s3-secret-key <secret-key>  --snapshot rke1db.zip && sleep 9223372036854775807"
+```
```

Comment:
```suggestion
2. Now you can either run mgiration agent directly on the node, or use the following [mainfest](https://github.com/rancher/migration-agent/blob/master/deploy/daemonset.yaml) to deploy migration-agent as a daemonset on the RKE1 cluster. Before you run the manifest file you need to edit the file to include the information about the s3 snapshot of RKE1:
    ```
    command:
          - "sh"
          - "-c"
          - "migration-agent --s3-region <region> --s3-bucket hgalal-s3 --s3-folder <s3 folder> --s3-access-key <access-key> --s3-secret-key <secret-key>  --snapshot rke1db.zip && sleep 9223372036854775807"
    ```
```

Reply:
so the problem is there is no combination of daemonset + job, meaning that you cant run a job on all nodes, so the only way I found is to run daemonset with infinite sleep, so that it wont be re-created again

Reply:
That just works for this specific run though. Once the node is restarted will it run the command again when the daemonset pod is recreated on the node?

For an infinite sleep, you might use the pause/mkfifo trick that klipper-lb uses:
https://github.com/k3s-io/klipper-lb/blob/master/entry#L13-L16

Reply:
yeah it does that, once its completed it will retry to run again